src/Database/Adapter/MariaDB.php

```diff
@@ -770,23 +770,60 @@ public function deleteDocument(string $collection, string $id): bool
      *
      * @param string $collection
      * @param Query[] $queries
-     * @param int $limit
-     * @param int $offset
-     * @param array $orderAttributes
-     * @param array $orderTypes
-     * @param array $cursor
-     * @param string $cursorDirection
      * @return Document[]
      * @throws Exception 
      * @throws PDOException 
      */
-    public function find(string $collection, array $queries = [], int $limit = 25, int $offset = 0, array $orderAttributes = [], array $orderTypes = [], array $cursor = [], string $cursorDirection = Database::CURSOR_AFTER): array
+    public function find(string $collection, array $queries = []): array
     {
         $name = $this->filter($collection);
         $roles = Authorization::getRoles();
         $where = ['1=1'];
         $orders = [];
 
+        $filters = [];
+        $limit = 25;
+        $offset = 0;
+        $orderAttributes = [];
+        $orderTypes = [];
+        $cursor = null;
+        $cursorDirection = Database::CURSOR_AFTER;
+        foreach ($queries as $query) {
+            if (!$query instanceof Query) continue;
+
+            $method = $query->getMethod();
+            $attribute = $query->getAttribute();
+            $values = $query->getValues();
+            switch ($method) {
+                case Query::TYPE_ORDERASC:
+                case Query::TYPE_ORDERDESC:
+                    if (!empty($attribute)) {
+                        $orderAttributes[] = $attribute;
+                    }
+
+                    $orderTypes[] = $method == Query::TYPE_ORDERASC ? Database::ORDER_ASC : Database::ORDER_DESC;
```

Comment:
```suggestion
                    $orderTypes[] = $method === Query::TYPE_ORDERASC ? Database::ORDER_ASC : Database::ORDER_DESC;
```

in PHP I would do a strict equal everytime, otherwise you might get type juggling 👍🏻

Reply:
just as an example if you didn't know:
```php
var_dump("0000" == 0);      // bool(true)
var_dump("0e12345" == "0"); // bool(true)
var_dump("0e34983" == 0);   // bool(true)
```

ref: https://www.php.net/manual/en/language.types.type-juggling.php

Reply:
Makes sense! Thanks for the tip! 🙏

Reply:
updated!